tests/org.eclipse.ui.tests/plugin.xml

```diff
@@ -3411,7 +3411,56 @@
        </markerContentGenerator>
         <markerContentGenerator
              id="org.eclipse.ui.tests.customScopeContentGenerator">
-       </markerContentGenerator>       
+       </markerContentGenerator>
+        <markerField
```

Comment:
> My initial expectation when screening this PR was that the Problems view would make the new columns available for addition, when going in the view options.

I admit, I had the same expectation (new columns would have to be added via the 'Configure Columns' dialog, but I did not test if that's really the case. I think, we should change that so that the users decide what the want to add to the problems view / markers view.

Reply:
What surprises me looking on this PR is the absence of new API added => means, it looks you use existing extensions point data to show some potentially already existing extensions in the Problems view that were hidden before.

If this assumption is correct, then extension point should be updated with the extra argument "yes, that contribution should be shown in the Problems view".

Otherwise, existing code will cause unexpected data to be shown that wasn't ever supposed to be shown.

Reply:
BTW, there is also "Markers" view too, and it might show more content as Problems view by default. 
So it should be clear for the extension point in which of the views the content is supposed to be shown.

Reply:
I did not change the existing API, that's correct. In theory, someone could have implemented marker support extensions with additional marker fields or types that are not usable at the moment and would become usable after merging this PR. (I think, this is very unlikely.) Would that be a problem if each user would still have to explicitly add these new fields to the list of visible fields in the problems or marker view? (I think, I should adapt this PR to let the user explicitly add the new fields to the views. I hope that can be done easily.)

Reply:
OK, having looked into the change. It seems quite odd that `ContentGeneratorDescriptor` lists the extensions `markerContentGeneratorExtension`. I would have expecte code that cares about marker content to check the extensions, on top of the "main" content. However:

1. There is no direct API to iterate over extensions of `ContentGeneratorDescriptor`.
2. `ContentGeneratorDescriptor.getFilterReferences()` goes over the extensions of the descriptor

So the PR does seem to simply add missing functionality.

Also odd to me is that `ContentGeneratorDescriptor.getFilterReferences()` does no caching, unlike the other methods. I assume this was the reason why @travkin79 removed the existing caching. From my POV, we should keep the caching. The main reason for the lack of caching is, I assume, `ContentGeneratorDescriptor.removeExtension()`. That one is likely called dynamically. `ContentGeneratorDescriptor.addExtensions()` is called (so far) only when constructing the descriptor.

So, if we want to actually be correct, I believe we must cache more dynamically. Possibly on the add and remove methods... I think for the `removeExtension` method this is probably necessary... The removes would be coming from: `ExtensionTracker.registryChanged()`

I'll look into this. In the worst case we cache as in the previous state in the PR (which I changed). I.e. cache the descriptor contributions, go over the extensions on-the-fly (no idea how costly that would be, its already done for the filters though... so nothing new).

Essentially, the `plugin.xml` glue is this:

```
        <markerContentGeneratorExtension
              generatorId="org.eclipse.ui.ide.problemsGenerator"
              id="org.eclipse.ui.tests.additionalProblemMarkerContentGenerator">
```

With the extension being:

```
       <markerContentGenerator
              id="org.eclipse.ui.tests.additionalProblemMarkerContentGenerator"
              name="Additional Problem Marker Content Generator">
           <markerFieldReference
           	     visible="true"
                 id="org.eclipse.ui.tests.markerField.problemKey">
```

This tells Eclipse, which view will be extended (the `Problems` view in this case) and with what.

If an existing contribution results in a column that is now visible, but this is not expected, that contribution should be adjusted to use `visible="false"`.

Though I have doubts that there are such contributions, considering that so far they do nothing.

Reply:
@travkin79  in the PR description / commit message you mention that groups are also touched by this change. Yet in the previous state of the PR I didn't see any change for group code?